### PR TITLE
[MAINTENANCE] Repo Cleanup: Move docs related files under docs dir

### PR DESCRIPTION
**Pull request by anthonyburdi**
Changes proposed in this pull request:
- Move docs related files into docs/ dir instead of the repo root and reorganize.
- Please see https://63f8f49f4f56480008a9a46e--anthony-docs-base-dir-change-testing.netlify.app/docs/ for a deploy preview. I've done some review but please check this preview for any issues before we merge this PR!
- This PR doesn't address `docs_rtd` and `readme_assets`. Those will be handled in future work.
- To coincide with this PR merging, we will also need to change the build settings on the main site to the following:
image

### Definition of Done

- [x] My code follows the Great Expectations [style guide](https://docs.greatexpectations.io/docs/contributing/style_guides/code_style)
- [x] I have performed a [self-review](https://docs.greatexpectations.io/docs/contributing/contributing_checklist) of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have run any local integration tests and made sure that nothing is broken.

### CODE REVIEW

**Review by Shinnnyshinshin — APPROVED** [review]
Thank you very much for this @anthonyburdi LGTM pending approval of the `sidebar.js` file.

**Review by donaldheppner — APPROVED** [review]

**Pull request merged**
